### ISSUE TITLE
Upcoming change to FullStory Data Export API on 7/31/18

### ISSUE THREAD

**Issue by ian0608** [singer-io/tap-fullstory#2]
Hello from FullStory! It’s awesome to see FullStory represented as a Singer tap.

We wanted to make you aware that the base endpoint used by this tap (https://www.fullstory.com/api/v1/export/) will be shutting down on July 31st.

A while back, we transitioned to a new Data Export endpoint (https://export.fullstory.com/api/v1/export/). This is the endpoint that’s reflected in our [Knowledge Base](https://help.fullstory.com/develop-rest/data-export-api). So far we’ve been supporting both the new and old endpoints, but we’ll be sunsetting the old endpoint on 7/31. This tap will need to switch to https://export.fullstory.com/api/v1/export to continue functioning after that date. The API functionality remains the same- only the URL is different.

Thanks!
Ian

**Comment by dmosorast** [comment]
@ian0608 Thanks for the heads up! I'm looking at this right now, and it seems that the new endpoint is no longer using `Content-Type` to specify if an export is GZip encoded (used [here](https://github.com/singer-io/tap-fullstory/blob/master/tap_fullstory/__init__.py#L105) in the tap). 

When the tap makes a request for `https://export.fullstory.com/api/v1/export/get?id=...`, the headers in the response are as follows:

```
ipdb> resp.headers
{'Date': 'Fri, 06 Jul 2018 14:01:10 GMT', 'Content-Type': 'application/json', 'Content-Disposition': 'attachment; filename=DataExport.json.gz', 'Content-Length': '26'}
```

The filename indicates it's a gzip file, but I'd rather not change the code to check for file extensions. Can the `Content-Type` be fixed? Where is the proper place to submit this feedback?

**Comment by ian0608** [comment]
@dmosorast Thanks for bringing this up! We discussed internally, and we'll be changing our headers to match the below. This change should go live mid- next week.

If a client does _not_ set the `Accept-Encoding: gzip` header on the request, we'll set the following headers on the response (this is the part being changed):
```
Content-Type: application/gzip
Content-Encoding: identity
Content-Disposition: attachment; filename=DataExport.json.gz
```
If a client _does_ set `Accept-Encoding: gzip`, the response headers will look like this (this is already the existing behavior and won't change):
```
Content-Type: application/json
Content-Encoding: gzip
Content-Disposition: attachment; filename=DataExport.json
```
This should play nicely with the Singer tap code. As a side note... I though Python `requests` would set `Accept-Encoding: gzip` automatically, and also automatically decode the response. Is there a reason this isn't being leveraged in the tap?

**Comment by ian0608** [comment]
@dmosorast Ah, good catch. I definitely see the `Accept-Encoding: gzip` header being set when I use `requests` locally, so there must be some other weirdness going on. Regardless, I think we should be all set here once our code change goes live. I can close the issue at that time. Thanks again!

**Comment by ian0608** [comment]
@dmosorast This change has been deployed- please let me know if you run into any issues.

**Comment by ian0608** [comment]
On second thought, probably best to leave the issue open until the endpoints are changed in the tap code :)

**Comment by dmosorast** [comment]
Thanks, @ian0608! It looks like the tap code is working now. I'll make a PR to update the endpoint and reference/close this issue once it's merged.

**Comment by dmosorast** [comment]
Merged in #3
